### PR TITLE
Cleanup MCU links

### PR DESCRIPTION
**Pull request by dotcypress**
Actualize all product links

### CODE REVIEW

File: stm32_part_table.yaml [hunk 1]
```diff
@@ -119,69 +119,69 @@ stm32f2:
...
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32f303.html
     rm: RM0316
     rm_title: STM32F303xB/C/D/E, STM32F303x6/8, STM32F328x8, STM32F358xC, STM32F398xE
     rm_url: https://www.st.com/resource/en/reference_manual/dm00043574.pdf
     members:
       - STM32F303
 
   stm32f373:
-    url: https://www.st.com/content/st_com/en/products/microcontrollers/stm32-32-bit-arm-cortex-mcus/stm32-mainstream-mcus/stm32f3-series/stm32f373.html
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32f3x8.html
```
Comment (adamgreig) [inline stm32_part_table.yaml]:
```suggestion
    url: https://www.st.com/en/microcontrollers-microprocessors/stm32f373.html
```

File: stm32_part_table.yaml [hunk 2]
```diff
@@ -119,69 +119,69 @@ stm32f2:
...
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32f334.html
     rm: RM0364
     rm_title: STM32F334xx
     rm_url: https://www.st.com/resource/en/reference_manual/dm00093941.pdf
     members:
       - STM32F334
 
   stm32f3x8:
-    url: https://www.st.com/content/st_com/en/products/microcontrollers/stm32-32-bit-arm-cortex-mcus/stm32-mainstream-mcus/stm32f3-series/stm32f3x8.html
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32f373.html
```
Comment (adamgreig) [inline stm32_part_table.yaml]:
```suggestion
    url: https://www.st.com/en/microcontrollers-microprocessors/stm32f3x8.html
```

File: stm32_part_table.yaml (hunk 1 above)
Comment (dotcypress) [reply stm32_part_table.yaml]: Done

File: stm32_part_table.yaml (hunk 2 above)
Comment (dotcypress) [reply stm32_part_table.yaml]: Done

**Pull request merged**